### ISSUE TITLE
 INVALID_ARGUMENT : Invalid Feed Input Name:token_type_ids

### ISSUE THREAD

**Issue by jianqianzhou** [microsoft/onnxruntime#6389]
output_model_path = "chinese_roberta_l-12_H-768.onnx"
session = onnxruntime.InferenceSession(output_model_path, sess_options, providers=['CPUExecutionProvider'])
# Warm up with one run.
results = session.run(None, inputs_onnx)
# Measure the latency.
start = time.time()
for _ in range(total_runs):
    results = session.run(None, inputs_onnx)
end = time.time()
print("ONNX Runtime cpu inference time for sequence length {} (model not optimized): {} ms".format(num_tokens, format((end - start) * 1000 / total_runs, '.2f')))
del session

  File "/work/tensor1.12/tf2.0/lib/python3.6/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 124, in run
    return self._sess.run(output_names, input_feed, run_options)
onnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Invalid Feed Input Name:token_type_ids

**Comment by gwang-msft** [comment]
More information is needed to look into the error.
If it is possible, please share the model and the complete inferencing code.

**Comment by jianqianzhou** [comment]
> More information is needed to look into the error.
> If it is possible, please share the model and the complete inferencing code. Otherwise this issue is non-actionable.

from transformers import BertTokenizer, TFBertForSequenceClassification
import tensorflow as tf

tokenizer = BertTokenizer.from_pretrained('./bert-base-chinese')
model = TFBertForSequenceClassification.from_pretrained('./bert-base-chinese')

inputs = tokenizer("rch能保证是不是回因为环境差异带来运行错误。", return_tensors="tf")
logits = model(inputs)

import keras2onnx
onnx_model = keras2onnx.convert_keras(model, model.name)
output_model_path = "chinese_roberta_l-12_H-768.onnx"
keras2onnx.save_model(onnx_model, output_model_path)

**Comment by gwang-msft** [comment]
The error message indicated that the input name "token_type_ids" does not match any input definition of the onnx model, please check if the onnx model has the same set of input names as your input.

**Comment by snnn** [comment]
I'll close it as it is stale.

**Comment by danielbellhv** [comment]
I have a much similar [bug](https://github.com/microsoft/onnxruntime/issues/10342#issuecomment-1018417578) as OP

**Comment by vel0c3t** [comment]
I had a similar bug. ONNX discarded _forward_ function parameters that **were not part of the calculation** of the output.

A way to check that your saved ONNX indeed has the inputs:
onnx_model = onnx.load("model.onnx")
onnx_model.graph.input

**Comment by satishsilveri** [comment]
can we do something like this:

```
from onnxruntime.capi.onnxruntime_pybind11_state import InvalidArgument

outputs = None
try:
      outputs = self.model.run(None, onnx_inputs)
except InvalidArgument as e:
      if 'token_type_ids' in str(e):
          onnx_inputs.pop('token_type_ids', None)

          outputs = self.model.run(None, onnx_inputs)
```

this is just a rough draft but we can add or make it better?

python file: modeling_ort.py
line number: 996

**Comment by Tanxj** [comment]
If you've declared an input variable in the argument list, you have to use it in your code, or you'll get this error. The unused variable is not necessarily the one given in the error message.
如果你在参数列表中声明了一项输入，你就必须要使用它，否则你就会遇到这个错误。闲置的变量未必是错误信息中提示的那一个。
